packages/studio/src/components/AssetSelector.tsx

```diff
@@ -96,15 +103,47 @@ export const AssetSelector: React.FC = () => {
 		[setAssetFoldersExpanded],
 	);
 
+	const onDragOver: React.DragEventHandler<HTMLDivElement> = useCallback(
+		(e) => {
+			e.preventDefault();
+		},
+		[],
+	);
+
+	const onDrop: React.DragEventHandler<HTMLDivElement> = useCallback(
+		(e) => {
+			e.preventDefault();
+			e.stopPropagation();
+			handleUploadFile(e.dataTransfer.files[0], dropLocation || '/');
+			setDropLocation(null);
```

Comment:
If the Studio server is disconnected or you deploy the Studio statically, the drop handler is enabled and fails with a generic error.

It should not accept dragging at all.

Use the `readOnly` props and the `const connectionStatus = useContext(StudioServerConnectionCtx)
		.previewServerState.type;` to determine if the Studio is connected to a server.